nixos/modules/services/misc/plex.nix

```diff
@@ -133,6 +133,23 @@ in
         KillSignal = "SIGQUIT";
         PIDFile = "${cfg.dataDir}/Plex Media Server/plexmediaserver.pid";
         Restart = "on-failure";
+
+        # Hardening
+        NoNewPrivileges = true;
+        PrivateTmp = true;
+        PrivateDevices = true;
```

Comment:
my only consern is hit one, which may hamper hardware accelerated transcoding

Reply:
So is there a change which can be done to address most use cases? It's trivial doe people override to the options manually, so potentially a note on the service could be sufficient.

Reply:
Assuming VA-API we could have `PrivateDevices = false; DeviceAllow = [ "/dev/dri/renderD128" ];`, but this does not account for CUDA, AMD/ROCm or Darwin. It's better not set IMO

Reply:
OK. I think I the best option is to add another option, one which allows people to enable those options with a flag. So `enableCuda`, `enableVAAPI`, `enableDarwin`, and `enableAMD`. It could also be a list, but separate enables have advantages. I can only test with VAAPI and CUDA, so I would need help with the others.